contracts/utils/Whitelist.sol

```diff
@@ -0,0 +1,141 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity >=0.6.0 <0.8.0;
+
+import "@openzeppelin/contracts/access/AccessControl.sol";
+
+/**
+    @title Whitelist
+    @dev Allows developers to implement whitelisting functionality
+    @dev Inherits methods and variables from AccessControl
+    @dev WHITELISTER_ROLE is used to grant permissions for whitelist control
+
+    Use `onlyWhitelisted` modifier for a method in order to restrict access
+    only for whitelisted addresses:
+
+    ```
+        contract MyContract is Whitelist {
+            function deposit() external onlyWhitelisted {
+                // allow execution only to addresses that are in the whitelist
+            }
+        }
+    ```
+
+    Addresses can be added and removed dynamically via the `addToWhitelist`
+    and `removeFromWhitelist` functions. You can check if an address is
+    whitelisted or not by calling `isWhitelisted` function.
+
+    The entire whitelist functionality can be enabled or disabled via the
+    `enableWhitelist` and `disableWhitelist` functions. You can check if the
+    feature is enabled by calling `isWhitelistEnabled` function.
+
+    The following events are emitted: Enabled, Disabled, AddressAdded, AddressRemoved
+*/
+
+contract Whitelist is AccessControl {
+    // See @openzeppelin/contracts/utils/EnumerableSet.sol for more info
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    /// @dev Emitted when whitelist feature is enabled
+    event Enabled(address senderAddress);
+    /// @dev Emitted when whitelist feature is disabled
+    event Disabled(address senderAddress);
+    /// @dev Emitted when an address is added to the whitelist
+    event AddressAdded(address addedAddress);
+    /// @dev Emitted when an address is removed from the whitelist
+    event AddressRemoved(address removedAddress);
+
+    /// @notice Whitelister role is used for whitelist administration
+    bytes32 public constant WHITELISTER_ROLE = keccak256("WHITELISTER_ROLE");
+
+    bool private _isWhitelistEnabled;
+    EnumerableSet.AddressSet private _whitelist;
```

Comment:
Why use this? If you want the list, you can read the events.

A mapping to bool should be enough and much cheaper in gas.

Reply:
[Replaced](https://github.com/Cerebellum-Network/chainbridge-solidity/pull/6/files#diff-d1d675d04151a0a3ec00bad5e7e2fbea4c4d2e70a75a00f656df8fa08553f5f7R50) EnumerableSet with `mapping`